src/EditorFeatures/VisualBasic/LineCommit/CommitBufferManager.vb

```diff
@@ -127,26 +127,38 @@ Namespace Microsoft.CodeAnalysis.Editor.VisualBasic.LineCommit
+                    Dim endLineNumber As Integer
+                    Dim endCharIndex As Integer
+                    Dim useSemantics = info._useStatementSemanticFormatting AndAlso Not isExplicitFormat
                     If useSemantics AndAlso Not isExplicitFormat Then
                         ' Avoid using semantics for formatting extremely large dirty spans without an explicit request
                         ' from the user. The "large span threshold" is 7000 lines. The 7000 line threshold is an
                         ' estimated value accounting for a lower-bound of the algorithmic complexity of text
                         ' differencing in designer cases along with measurements of a pathological example demonstrated
                         ' at 14000 lines. We expect Windows Forms designer formatting operations to run in under ~15
                         ' seconds on average current hardware when nearing the threshold.
-                        Dim startLineNumber = 0
-                        Dim startCharIndex = 0
-                        Dim endLineNumber = 0
-                        Dim endCharIndex = 0
-                        info.SpanToFormat.GetLinesAndCharacters(startLineNumber, startCharIndex, endLineNumber, endCharIndex)
+                        info._statementSpanToSemanticFormat.GetLinesAndCharacters(startLineNumber, startCharIndex, endLineNumber, endCharIndex)
                         If endLineNumber - startLineNumber > 7000 Then
                             useSemantics = False
                         End If
+                        tree = _dirtyState.BaseDocument.GetSyntaxTreeSynchronously(cancellationToken)
+                        _commitFormatter.CommitRegion(info._statementSpanToSemanticFormat, isExplicitFormat, useSemantics, dirtyRegion, _dirtyState.BaseSnapshot, tree, cancellationToken)
                     End If
 
-                    Dim tree = _dirtyState.BaseDocument.GetSyntaxTreeSynchronously(cancellationToken)
-                    _commitFormatter.CommitRegion(info.SpanToFormat, isExplicitFormat, useSemantics, dirtyRegion, _dirtyState.BaseSnapshot, tree, cancellationToken)
+                    useSemantics = info._useSemantics AndAlso Not info._useStatementSemanticFormatting
+                    If useSemantics AndAlso Not isExplicitFormat Then
+                        ' same comment at above
+                        info._spanToFormat.GetLinesAndCharacters(startLineNumber, startCharIndex, endLineNumber, endCharIndex)
+                        If endLineNumber - startLineNumber > 7000 Then
+                            useSemantics = False
+                        End If
+                    End If
+
+                    tree = _dirtyState.BaseDocument.GetSyntaxTreeSynchronously(cancellationToken)
+                    _commitFormatter.CommitRegion(info._spanToFormat, isExplicitFormat, useSemantics, dirtyRegion, _dirtyState.BaseSnapshot, tree, cancellationToken)
                 End Using
```

Comment:
all of tihs needs comments to explain what's going on. also, all the repetition strongly indicates that there should be a helper method here. that they share.